### DISCUSSION TITLE
TypeScript type hinting in vscode

### DISCUSSION THREAD

**Discussion by officialankan** [prisma/prisma#22983] [Q&A] [answered]
### Question

Hi, 

I'm new to full stack development and I only partake in my, limited, free time. It is certainly easier to dabble in this today than during my previous run of, limited, free time about 10 years ago.

I'm in VSCode and running a SvelteKit project with Supabase and using Prisma to handle my database schema and queries. 

Here is a a `Profile` model in my `schema.prisma`:

```prisma
model Profile {
  id                String             @id @default(uuid()) @db.Uuid
  firstName         String?            @map("first_name")
  lastName          String?            @map("last_name")
  createdAt         DateTime           @default(now()) @map("created_at")
  updatedAt         DateTime           @default(now()) @updatedAt @map("updated_at")
  }
```

and in a SvelteKit route I'm successfully retrieving this with:

```ts
// +page.server.ts
import prisma from "$lib/prisma";
import type { PageServerLoad } from "./$types";

export const load = (async () => {
    const response = await prisma.profile.findFirstOrThrow();
    return { profileData: response }
}) satisfies PageServerLoad
```

However, I was expecting some type hinting when doing some work with the result, but it is assigned the following type:

```ts
const response: $Result.GetResult<Prisma.$ProfilePayload<ExtArgs>, T, "findFirstOrThrow">
````

In my `+page.svelte` I can successfully work with my field data like:

```ts
// +page.svelte
<script lang="ts">
	import type { PageData } from './$types';

	export let data: PageData;
	let { profileData } = data;
	let firstName = profileData.firstName; // this works!
```

but I don't get any nice type hinting, I can only see 

![bild](https://github.com/prisma/prisma/assets/13569761/6c8cb175-1f6a-488a-a71c-440c2b7ab3b5)

when hovering in vscode....

- OS: ubuntu / WSL2
- Database: Postgres

Version:

prisma                  : 5.9.0
@prisma/client          : 5.9.0
Computed binaryTarget   : debian-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v20.9.0
Query Engine (Node-API) : libquery-engine 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.9.0-32.23fdc5965b1e05fc54e5f26ed3de66776b93de64
Default Engines Hash    : 23fdc5965b1e05fc54e5f26ed3de66776b93de64
Studio                  : 0.497.0

**Comment by ludralph** [comment, marked as answer]
Hi @officialankan 👋 

Thank you for raising this question.

There seems to have been some regression with types in our 5.9.0 release. Can you please update to the patch release of [5.9.1](https://github.com/prisma/prisma/releases/tag/5.9.1) to see if it fixes the issue for you.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by officialankan** [reply]
That did it! Thanks for the quick reply.

**Reply by ludralph** [reply]
You're welcome 🙏

**Comment by ryanoillataguerre** [comment]
Fwiw, I had this same problem. Upgrading to 5.9.1 fixed it for me too
